icu4c/source/test/intltest/units_test.cpp

```diff
@@ -400,97 +401,187 @@ void UnitsTest::testConverterWithCLDRTests() {
     const MeasureUnitImpl &outputImpl3 = MeasureUnitImpl::forMeasureUnit(output, tempOutput, status);
-    converter = ComplexUnitsConverter(inputImpl3, outputImpl3, rates, status);
-    measures = converter.convert((2.0 - DBL_EPSILON), nullptr, status);
-    assertEquals("measures length", 2, measures.length());
-    if (2 == measures.length()) {
-        assertEquals("light-year test: measures[0] value", 2.0,
-                     measures[0]->getNumber().getDouble(status));
-        assertEquals("light-year test: measures[0] unit", MeasureUnit::getLightYear().getIdentifier(),
-                     measures[0]->getUnit().getIdentifier());
-        assertEquals("light-year test: measures[1] value", 0.0,
-                     measures[1]->getNumber().getDouble(status));
-        assertEquals("light-year test: measures[1] unit", MeasureUnit::getMeter().getIdentifier(),
-                     measures[1]->getUnit().getIdentifier());
-    }
+    auto converter = ComplexUnitsConverter(inputImpl3, outputImpl3, rates, status);
+    status.assertSuccess();
 
     // 1e-15 light years is 9.46073 meters (calculated using "bc" and the CLDR
     // conversion factor). With double-precision maths, we get 10.5. In this
-    // case, we're off by almost 1 meter.
+    // case, we're off by a bit more than 1 meter.
     measures = converter.convert((1.0 + 1e-15), nullptr, status);
     assertEquals("measures length", 2, measures.length());
     if (2 == measures.length()) {
         assertEquals("light-year test: measures[0] value", 1.0,
                      measures[0]->getNumber().getDouble(status));
         assertEquals("light-year test: measures[0] unit", MeasureUnit::getLightYear().getIdentifier(),
                      measures[0]->getUnit().getIdentifier());
-        assertEqualsNear("light-year test: measures[1] value", 10,
-                         measures[1]->getNumber().getDouble(status), 1);
-        assertEquals("light-year test: measures[1] unit", MeasureUnit::getMeter().getIdentifier(),
-                     measures[1]->getUnit().getIdentifier());
-    }
-
-    // 2e-16 light years is 1.892146 meters. We consider this in the noise, and
-    // thus expect a 0. (This test fails when 2e-16 is increased to 4e-16.)
-    measures = converter.convert((1.0 + 2e-16), nullptr, status);
-    assertEquals("measures length", 2, measures.length());
-    if (2 == measures.length()) {
-        assertEquals("light-year test: measures[0] value", 1.0,
-                     measures[0]->getNumber().getDouble(status));
-        assertEquals("light-year test: measures[0] unit", MeasureUnit::getLightYear().getIdentifier(),
-                     measures[0]->getUnit().getIdentifier());
-        assertEquals("light-year test: measures[1] value", 0.0,
-                     measures[1]->getNumber().getDouble(status));
+        assertEqualsNear("light-year test: measures[1] value", 9.46073,
+                         measures[1]->getNumber().getDouble(status), 1.5);
```

Comment:
I decided to use the more precise measurement as "expected", and have the full inacuracy, relative to that, in the "accuracy" field.